src/communication/expr.rs

```diff
@@ -0,0 +1,308 @@
+
+                callee_method.component.path != class.component.component.path
+                    && service_name_equiv
+                    && endpoint_config_equiv
+            };
+
+        // Search through the modules pertaining to the client using the client name
+        if is_communication_call(&client_ident, method_ident, &client_name) {
+            for module in modules.iter() {
+                for class in module.classes.iter() {
+                    for method in class.component.methods.iter() {
+                        if found_rpc_endpoint(class, method, method_ident)
+                            || found_rest_endpoint(class, method, &client_name)
+                        {
+                            // Found the endpoint definition
+                            endpoint_call = Some(EndpointCallExpr::new(
+                                module.module_name.clone(),
+                                Some(class.component.container_name.clone()),
+                                method.method_name.clone(),
+                                self.clone(),
+                            ));
+                            break;
+                        }
+                    }
+                    if class
+                        .component
+                        .container_name
+                        .to_lowercase()
+                        .contains(&client_name)
+                        && endpoint_call.is_some()
+                    {
+                        // Prefer this method over any previously chosen ones because
+                        // we found a closely named Service module and updated the endpoint call.
+                        return map_endpoint_call(endpoint_call);
+                    }
+                }
+            }
+        }
+
+        map_endpoint_call(endpoint_call)
+    }
+}
+
+fn match_ident_or(expr: &Expr) -> Option<&Ident> {
+    match expr {
+        Expr::Ident(ident) => Some(ident),
+        _ => None,
+    }
+}
+
+fn get_rpc_service_name(
+    class: Option<&ClassOrInterfaceComponent>,
+    client_ident: &str,
+) -> Option<String> {
+    // Search through the modules pertaining to the client using the client name
+    let mut client_name = client_ident
+        .replace("client", "")
+        .replace("pool", "")
+        .replace("_", "");
+    if let Some(class) = class {
+        for field in class.field_components.iter() {
+            // Go by type information for finding the service name. If not available
+            // use naming convention.
+            let field_name = if field.r#type != "" {
+                field.r#type.to_lowercase()
+            } else {
+                field.field_name.to_lowercase()
+            };
+
+            if field_name.contains("client") {
+                // Strip excess type information and unneeded naming like "client" and underscores
+                let mut field_name = if field_name.contains("client>") {
```

Comment:
Why do we need to treat `client` and `client>` differently, since it seems we're using the first item in the split result and thus splitting on `client` would generate the right one for both? Or am I just misreading this code?

Reply:
It's because in Go there won't be a > at the end so the split will fail since it can't find that string

Reply:
My bad, I was referring to just splitting on `client` instead of on `client>`. Not a super big deal either way, though, it's perfectly readable and the performance impact would be imperceptible.

Reply:
Oh I see what you mean now. Yeah that could also be done and would clean the code up a bit